lnrpc/routerrpc/router_server.go

```diff
@@ -454,55 +454,155 @@ func (s *Server) GetMissionControlConfig(ctx context.Context,
 	req *GetMissionControlConfigRequest) (*GetMissionControlConfigResponse,
 	error) {
 
+	// Query the current mission control config.
 	cfg := s.cfg.RouterBackend.MissionControl.GetConfig()
-	eCfg, ok := cfg.Estimator.Config().(*routing.AprioriConfig)
-	if !ok {
-		return nil, fmt.Errorf("unknown estimator config type")
-	}
-
-	return &GetMissionControlConfigResponse{
+	resp := &GetMissionControlConfigResponse{
 		Config: &MissionControlConfig{
-			HalfLifeSeconds: uint64(
-				eCfg.PenaltyHalfLife.Seconds()),
-			HopProbability: float32(
-				eCfg.AprioriHopProbability,
-			),
-			Weight:                float32(eCfg.AprioriWeight),
 			MaximumPaymentResults: uint32(cfg.MaxMcHistory),
 			MinimumFailureRelaxInterval: uint64(
 				cfg.MinFailureRelaxInterval.Seconds(),
 			),
 		},
-	}, nil
+	}
+
+	// We only populate fields based on the current estimator.
+	switch v := cfg.Estimator.Config().(type) {
+	case routing.AprioriConfig:
```

Comment:
Also this commit would benefit from first making all the config restructuring changes for apriori. Maybe even in the first half of the commit stack don't even mention bimodal anywhere. And then when everything is in place, add the bimodal commits.

Reply:
I would have chosen to indeed regenerate twice for the sake of getting a clean commit that adds bimodal, but of course no blocker.